### PR TITLE
[WIP] Tabs UI

### LINKED ISSUE #61: Rework the styling of the Syngrafias frontend
It needs to look good. Very good.

### PR DESCRIPTION
**Pull request by shivangswain**
## Description

Adds the tabbed UI. Will resolve #61.

**Have you read the [Contributing Guidelines](https://github.com/t0xic0der/syngrafias/blob/master/CONTRIBUTING.md) before opening this PR**
Yes

## Screenshot
image

---

## Things to do

- [X] Add tab UI
- [X] Make tabs scrollable
- [X] Monospace tabs text
- [X] Remove sticky menu
- [X] Make editor full height
- [X] Add copy button next to Workspace ID
- [X] Add X to close on each tab

### CODE REVIEW

**Comment by t0xic0der** [comment]
@shivangswain At the end of the PR, remove the sticky nature of the menubar as the page would no longer be scrollable. The sticky property would not be needed anymore and it also messes up #88.

**Comment by t0xic0der** [comment]
@shivangswain Check the following image out.

image

You need to add the copy button too at the menu bar.

**Review by gridhead — CHANGES_REQUESTED** [review]
When clicked on the REMOVE CELL icon (The 'X' icon I mean) - The following error shows up. Please look into it.

image

**Comment by t0xic0der** [comment]
@VaibhavSaini19 @s-katte Please go ahead and review this plocs.

**Comment by t0xic0der** [comment]
Here's the preview.

image

File: static/jscn/syngrafias.js [hunk 1]
```diff
@@ -325,23 +337,26 @@ function recvpush(celliden, username) {
 }
 
 function makecell(celliden) {
+    $("#tabhead").append(`
+        <div class='link item' id='tabiden-${celliden}' data-tab='id-${celliden}' onclick='activatetab("${celliden}")'><span class="monotext">${celliden}&nbsp;<i id='close-${celliden}' class='link times icon' style='margin: 0;' onclick='sendpull("${celliden}");')></i></span></div>`
+    );
```
Comment (VaibhavSaini19) [inline static/jscn/syngrafias.js:342]: Could you format this with correct indentation please

File: static/jscn/syngrafias.js [hunk 2]
```diff
@@ -354,6 +369,20 @@ function makecell(celliden) {
         minSize: [150, 150],
```
Comment (VaibhavSaini19) [inline static/jscn/syngrafias.js:369]: The minimum size for the Split functionality isn't working on my end

**Review by VaibhavSaini19 — CHANGES_REQUESTED** [review]
Bugs mentioned in the comments above

File: static/jscn/syngrafias.js (hunk 1 above)
Comment (t0xic0der) [reply static/jscn/syngrafias.js:342]: > Also, it would be great if you could format this code with proper indentation as well

+1 to this

**Review by s-katte — CHANGES_REQUESTED** [review]
Few changes requested:

- [ ] Put a cross on the top-right corner of Usage Tips to remove it if not required.
- [ ] In tabs UI, if we open multiple tabs, and have a content in one tab with overflow-y (_just the point when scroll bar comes up_), the height of tabs section changes, improve this.

filled with content
image  

without content
image

- [ ] same is the issue if we toggle the split view --> when only preview is open, height reduces by some pixels.
- [ ] remove extra padding in the name of tabs to use the extra space for the content writing
- [ ] Add horizontal padding for the tabs, it does not align with rest ui
- [ ] Here is another major bug
image
- [ ] After the file is opened followed by parse, on clicking continue, the modal persists and content is not loaded properly, do test that once again.
- [ ] The cell is there, but still unable to delete it
image
- [ ] After deleting a tab, the focus must be on the last tab, instead of none.
- [ ] In tabs UI, on clicking ```i```, for information, the title should be ```Tab Information```, instead of ```Cell Information```

**Comment by t0xic0der** [comment]
The documents do not open up.

image

There is no option to switch over to the singular mode.

Clicking on the `i` icon gets me more console errors.

image

Condense the tab padding and font sizes.

image

**Pull request closed without merging**